app/controllers/locations_controller.rb

```diff
@@ -141,4 +170,11 @@ def set_coordinates
     end
   end
 
+  def match_keyword_to_category(keyword)
```

Comment:
👍

Reply:
Thanks for the thorough feedback @anna-dorigo, I'll get started on those fixes/cleanup 😊